src/passes/StackCheck.cpp

```diff
@@ -145,34 +142,23 @@ struct StackCheck : public Pass {
 
     Builder builder(*module);
 
-    // Add the globals.
-    auto stackBase =
-      module->addGlobal(builder.makeGlobal(stackBaseName,
-                                           stackPointer->type,
-                                           builder.makeConstPtr(0),
-                                           Builder::Mutable));
-    auto stackLimit =
-      module->addGlobal(builder.makeGlobal(stackLimitName,
-                                           stackPointer->type,
-                                           builder.makeConstPtr(0),
-                                           Builder::Mutable));
-
     // Instrument all the code.
     PassRunner innerRunner(module);
-    EnforceStackLimits(stackPointer, stackBase, stackLimit, builder, handler)
+    EnforceStackLimits(stackPointer, stackBase, stackEnd, builder, handler)
       .run(&innerRunner, module);
 
-    // Generate the exported function.
+    // TODO: Remove this function once emscripten stops calling it.
+    // This function is now a no-op, but it still needs to be here until
+    // emscripten is changed.
+    wasm::Name limitsFuncName = "__set_stack_limits";
```

Comment:
Can you put this back up the top to avoid churn?  (i.e. just add the comment to existing static?)